app/assets/javascripts/select_menu.js

```diff
@@ -0,0 +1,61 @@
+function initialize_select_menu() {
```

Comment:
The select menu has a JS so that it can be consistent with the search bar. The search bar only refreshes the partial and not the entire page. So we wanted to do the same with the search menu.

Reply:
Yes, I agree it's not ideal, and am very open to a different approach if necessary (though I couldn't think of any that wouldn't require changing the select menu in some way). The reasoning behind that is that to update the select menu links with the latest search query, re-rendering the select menu is necessary, which causes all existing jquery event listeners to be destroyed, hence the need to re-initialize all the jquery hooks. 

Other options I've thought of: 

1. Combine the select menu and search bar into a single form: `shared/_select-menu.html` as it is does not lend itself well to integration within existing forms, and I'd be open to exploring changing it but I want to make sure that we don't specialize this component so much it can't be used as a component in the general case

2. Re-render just the links in the search menu -- would require functional changes to _search_menu.js and probably refactoring the actual list links html into a separate partial so it can be re-renderd separately via ERB. 

3. Re-render the entire container partial containing the search bar / sort filter every submit -- icky, would require we re-hookup event listeners to both search and sort filter

Reply:
@d12 I should have also mentioned that we're using I believe(?) a primer.style select menu component similar to the rest of our components and not an actual native select menu, so that's why jquery is used in the first place to interact with it